source/sdk/lang/Array.h

```diff
@@ -12,20 +11,13 @@
 
 #include <stdint.h>
 
-
 #define _lang_array__Array_new(type, size) ((_lang_array__Array) { size, array_malloc((size) * sizeof(type)) });
 
 #if defined(safe)
 #define _lang_array__Array_get(array, index, type) ( \
-    (index < 0 || index >= array.length) ? \
-    lang_Exception__Exception_throw((lang_Exception__Exception *) lang_Exception__OutOfBoundsException_new_noOrigin(index, array.length)), \
-    *((type*) NULL) : \
-    ((type*) array.data)[index])
+    ((type*) array.data)[(index < 0 || index >= array.length) ? lang_Exception__Exception_throw((lang_Exception__Exception *) lang_Exception__OutOfBoundsException_new_noOrigin(index, array.length)), -1 : index])
```

Comment:
Previously the `_lang_array__Array_set` macro was causing confusion when rock was trying to resolve the types after macro expansion. Both versions should do the same thing, in C at least. But the error was thrown at the ooc stage, so I think rock did macro expansion too and failed.